src/main/java/org/broadinstitute/hellbender/tools/reference/CompareReferences.java

```diff
@@ -141,6 +171,28 @@ public void traverse(){
         for(ReferencePair pair : referencePairs){
             System.out.println(pair);
         }
+
+        if(referencePairs.size() != 1 && baseComparisonMode != BaseComparisonMode.NO_BASE_COMPARISON){
+            throw new UserException.BadInput("");
+        }
+        if(baseComparisonMode != BaseComparisonMode.NO_BASE_COMPARISON){
```

Comment:
You can move the `if(referencePairs.size() != 1)` check from above inside this `if(baseComparisonMode != BaseComparisonMode.NO_BASE_COMPARISON)` block as well